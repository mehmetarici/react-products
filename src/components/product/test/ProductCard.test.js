import { render } from '@testing-library/react';
import { ProductCard } from "../../index";

describe('<ProductCard />', () => {
    const props = {
        product: {
            "id": 1,
            "name": "Iphone 11 Kılıf",
            "imageName": "sample-product-3.png",
            "category": "Telefon Aksesuar",
            "brand": "Apple",
            "color": "Kırmızı",
            "stockCount": 4,
            "inBasket": false,
            "createdAt": "2022-01-01T10:30:00",
            "amount": {
                "currency": "TL",
                "original": 240,
                "discounted": 216,
                "discountRatePercentage": 10
            }
        }
    };

    const setup = (props) => {
        const utils = render(ProductCard(props));
        return {...utils};
    };

    it('renders correctly', () => {
        const {asFragment} = setup(props);
        expect(asFragment()).toMatchSnapshot();
    });
});
