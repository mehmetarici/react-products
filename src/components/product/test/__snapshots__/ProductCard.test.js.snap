// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="product"
  >
    <div
      class="product-image"
    >
      <img
        alt="Iphone 11 Kılıf"
        src="sample-product-3.png"
      />
    </div>
    <h2
      class="product-title"
    >
      Iphone 11 Kılıf
    </h2>
    <div
      class="product-info"
    >
      <span>
        Marka: 
        <span
          class="product-info-value"
        >
          Apple
        </span>
      </span>
      <span>
        Renk: 
        <span
          class="product-info-value"
        >
          Kırmızı
        </span>
      </span>
    </div>
    <div
      class="product-price"
    >
      <span
        class="price"
      >
        216.00 TL
      </span>
      <div
        class="discount"
      >
        <span
          class="normal-price"
        >
          240.00 TL
        </span>
        <span
          class="percentage"
        >
          10%
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
